torchdynamo/variables/torch.py

```diff
@@ -180,6 +182,12 @@ def call_function(
                 tensor_with_tf_override.subclass_torch_function__func,
                 tensor_with_tf_override.subclass_type,
             )
+        elif self.value is torch.autograd.profiler.record_function:
+            assert len(args) == 1
+            return ProfileRunModeVariable(str(args[0].as_proxy()), **options)
+        elif self.value is torch.autograd.profiler.profile:
+            # Passthrough
+            return ConstantVariable(torch.autograd.profiler.profile, **options)
```

Comment:
This isn't really a ConstantVariable.  I curious why this is needed.

Reply:
Good question. I don't know the correct way to make this pass-through, I want it to no-op. If it goes down to `return TensorVariable.create`, it will reach:

`assert (
                False
            ), f"torch.* op returned non-Tensor {typestr(example_value)} {proxy.node.op} {proxy.node.target}"
`
Is UserDefinedVariable more correct? or perhaps we need a new type here?

Reply:
Returning None actually works here, and is probably more correct. Thoughts?

Reply:
ConstantVariable() will turn it into a constant (evaluated at compile time), which I don't think is what we want.

If we want to put this into the graph, then we should modify `TensorVariable.create` to detect this function wrap the return value properly.  

I think we may just want to break the graph on this by adding it to the disallowed list in `allowed_functions.py`.